--- /dev/null
+++ b/Makefile
@@ -0,0 +1,23 @@
+VERSION=6.1.0
+
+DEST_ROOT=debian/jboss
+DEST=$(DEST_ROOT)/var/lib/jboss
+
+all:
+
+	true
+
+clean:
+
+install:
+	cp -r lib $(DEST)
+	cp -r jar-versions.xml $(DEST)
+	cp -r client $(DEST)
+	cp -r server $(DEST)
+	cp -r bin $(DEST)
+	cp -r common $(DEST)
+	cp -r usr $(DEST_ROOT)
+
+	# link jboss undeploy
+	ln -f -s jboss-deploy $(DEST_ROOT)/usr/bin/jboss-undeploy
+
