--- a/usr/bin/jboss-deploy
+++ b/usr/bin/jboss-deploy
@@ -57,7 +57,7 @@
 else
   echo downloading $URL
   echo --------
-  echo wget -nv -N --max-redirect=2 -P $DOWNLOAD_DIR $URL
+  echo wget -nv -N $OPT_FN --max-redirect=2 -P $DOWNLOAD_DIR $URL
   if wget -nv -N $OPT_FN --max-redirect=2 -P $DOWNLOAD_DIR $URL
   then
       NEW_FILE=`ls $DOWNLOAD_DIR/$ARTIFACT_ID* | tail -n1`
@@ -67,7 +67,7 @@
       then
           echo --------
           echo deploying $ARTIFACT to $TARGET
-          cp -f $DOWNLOAD_DIR/$ARTIFACT_ID* $TARGET
+          cp -f $DOWNLOAD_DIR/$NEW_FILE $TARGET
           chmod 644 $TARGET
       else
           echo --------
