--- a/debian/jboss.default
+++ b/debian/jboss.default
@@ -10,7 +10,9 @@
 DAEMON_OPTS=""
 
 
-JAVA_OPTS="-Xmx1024M -Xms1024M -XX:PermSize=128m -XX:MaxPermSize=256m -XX:+CMSClassUnloadingEnabled" 
+JAVA_OPTS="-Xmx1024M -Xms1024M -XX:PermSize=128m"
+JAVA_OPTS="$JAVA_OPTS -XX:MaxPermSize=256m -XX:+CMSClassUnloadingEnabled"
+JAVA_OPTS="$JAVA_OPTS -Djboss.server.log.dir=/var/log/jboss"
 export JAVA_OPTS
 
 # Profile that gets loaded. Must be set.
